File: Python/Program/Harddrive.py
from Komponen import Komponen

class Harddrive(Komponen):
    def __init__(self, kapasitas=0, tipe="", kecepatan_baca=0, kecepatan_tulis=0, nama="", harga=0, merk=""):
        super().__init__(nama, harga, merk)
        self.kapasitas = kapasitas
        self.tipe = tipe
        self.kecepatan_baca = kecepatan_baca
        self.kecepatan_tulis = kecepatan_tulis
    
    def set_kapasitas(self, kapasitas):
        self.kapasitas = kapasitas
    
    def set_tipe(self, tipe):
        self.tipe = tipe
    
    def set_kecepatan_baca(self, kecepatan_baca):
        self.kecepatan_baca = kecepatan_baca
    
    def set_kecepatan_tulis(self, kecepatan_tulis):
        self.kecepatan_tulis = kecepatan_tulis
    
    def get_kapasitas(self):
        return self.kapasitas
    
    def get_tipe(self):
        return self.tipe
    
    def get_kecepatan_baca(self):
        return self.kecepatan_baca
    
    def get_kecepatan_tulis(self):
        return self.kecepatan_tulis
    
    def format_drive(self):
        print(f"Formatting drive {self.get_nama()}...")
        print("Drive formatted successfully!")
    
    def get_info(self):
        super().get_info()
        print(f"Kapasitas: {self.kapasitas} GB, Tipe: {self.tipe}, Kecepatan Baca: {self.kecepatan_baca} MB/s, Kecepatan Tulis: {self.kecepatan_tulis} MB/s")

File: Java/Program/Harddrive.java
public class Harddrive extends Komponen {
    private int kapasitas;
    private String tipe;
    private int kecepatanBaca;
    private int kecepatanTulis;

    public Harddrive() {
    }

    public Harddrive(int kapasitas, String tipe, int kecepatanBaca, int kecepatanTulis, String nama, float harga, String merk) {
        super(nama, harga, merk);
        this.kapasitas = kapasitas;
        this.tipe = tipe;
        this.kecepatanBaca = kecepatanBaca;
        this.kecepatanTulis = kecepatanTulis;
    }

    public void setKapasitas(int kapasitas) {
        this.kapasitas = kapasitas;
    }

    public void setTipe(String tipe) {
        this.tipe = tipe;
    }

    public void setKecepatanBaca(int kecepatanBaca) {
        this.kecepatanBaca = kecepatanBaca;
    }

    public void setKecepatanTulis(int kecepatanTulis) {
        this.kecepatanTulis = kecepatanTulis;
    }

    public int getKapasitas() {
        return kapasitas;
    }

    public String getTipe() {
        return tipe;
    }

    public int getKecepatanBaca() {
        return kecepatanBaca;
    }

    public int getKecepatanTulis() {
        return kecepatanTulis;
    }

    public void formatDrive() {
        System.out.println("Formatting drive " + getNama() + "...");
        System.out.println("Drive formatted successfully!");
    }

    @Override
    public void getInfo() {
        super.getInfo();
        System.out.println("Kapasitas: " + kapasitas + " GB, Tipe: " + tipe + ", Kecepatan Baca: " + kecepatanBaca + " MB/s, Kecepatan Tulis: " + kecepatanTulis + " MB/s");
    }
}